---
layout: page
title: A workflow to unify the Known and Unknown
---
We implemented a computational workflow ([Agnostos](https://www.encyclopedia.com/environment/encyclopedias-almanacs-transcripts-and-maps/agnostos-theos)) to structure and explore the large pool of genes with unknown functions found in microbial genomes and metagenomes. We used a protein domain-based approach to partition more than 400 million predicted genes from 1,628 metagenomes and ~29 genomes into different categories of gene clusters of known and unknown functions.


| ![space-1.jpg](img/workflow.png){:class="img-responsive"} | 
|:--:| 
| **Agnostos workflow different modules** |



The AGNOSTOS WORKFLOW can be found [HERE](https://github.com/functional-dark-side/agnostos-wf).

<br>

The methods (included all the scripts/code) and results for each module are documented. Below you can find the links to the different reports:

[**DATA**](1_Data)


<h3 class="section-heading  text-primary">Workflow steps</h3>


-   [1. Gene prediction](2_Gene_prediction)
<br>

-   [2. MMseqs clustering](3_MMseqs_clustering)
<br>

-   [3. Pfam annotations](4_Pfam_annotation)
<br>

-   [4. Cluster validation](5_Cluster_validation)
<br>

-   [5. Cluster refinement](6_Cluster_refinement)
<br>

-   [6. Cluster classification in categories](7_Cluster_classification)
<br>

-   [7. Known-Unknown cluster category refinement](8_Known-Unknown_refinement)
<br>

-   [8. Cluster communities inference](9_Cluster_communities)

<br>
